src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -1470,6 +1471,327 @@ + (id)CHIPEncodeAndDecodeNSObject:(id)object
+                      completion:(MTRDeviceResponseHandler)completion
+{
+    endpointID = (endpointID == nil) ? nil : [endpointID copy];
+    clusterID = (clusterID == nil) ? nil : [clusterID copy];
+    eventID = (eventID == nil) ? nil : [eventID copy];
+    eventMin = (eventMin == nil) ? nil : [eventMin copy];
+    params = (params == nil) ? nil : [params copy];
+    auto * bridge = new MTRDataValueDictionaryCallbackBridge(queue, completion,
+        ^(ExchangeManager & exchangeManager, const SessionHandle & session, MTRDataValueDictionaryCallback successCb,
+            MTRErrorCallback failureCb, MTRCallbackBridgeBase * bridge) {
+            auto resultArray = [[NSMutableArray alloc] init];
+            auto resultSuccess = [[NSMutableArray alloc] init];
+            auto resultFailure = [[NSMutableArray alloc] init];
+            auto onSuccessCb = [resultArray, resultSuccess](
+                                   const app::ConcreteEventPath & eventPath, const MTRDataValueDictionaryDecodableType & aData) {
+                [resultArray addObject:@ {
+                    MTREventPathKey : [[MTREventPath alloc] initWithPath:eventPath],
+                    MTRDataKey : aData.GetDecodedObject()
+                }];
+                if ([resultSuccess count] == 0) {
+                    [resultSuccess addObject:[NSNumber numberWithBool:YES]];
+                }
+            };
+
+            auto onFailureCb = [resultArray, resultFailure](const app::ConcreteEventPath * eventPath, CHIP_ERROR aError) {
+                if (eventPath) {
+                    [resultArray addObject:@ {
+                        MTREventPathKey : [[MTREventPath alloc] initWithPath:*eventPath],
+                        MTRErrorKey : [MTRError errorForCHIPErrorCode:aError]
+                    }];
+                } else if ([resultFailure count] == 0) {
+                    [resultFailure addObject:[MTRError errorForCHIPErrorCode:aError]];
+                }
+            };
+
+            app::EventPathParams eventPath;
+            if (endpointID) {
+                eventPath.mEndpointId = static_cast<chip::EndpointId>([endpointID unsignedShortValue]);
+            }
+            if (clusterID) {
+                eventPath.mClusterId = static_cast<chip::ClusterId>([clusterID unsignedLongValue]);
+            }
+            if (eventID) {
+                eventPath.mEventId = static_cast<chip::EventId>([eventID unsignedLongValue]);
+            }
+            app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
+            CHIP_ERROR err = CHIP_NO_ERROR;
+
+            chip::app::ReadPrepareParams readParams(session);
+            [params toReadPrepareParams:readParams];
+            readParams.mpEventPathParamsList = &eventPath;
+            readParams.mEventPathParamsListSize = 1;
+            readParams.mEventNumber.SetValue(static_cast<chip::EventNumber>([eventMin unsignedLongLongValue]));
```

Comment:
Does this do the right thing when `eventMin` is nil?  I guess it does because we will get 0, but more correct would be to not set mEventNumber at all.

Reply:
We changed for Read Request to be sent without EventFilterIB if eventMin is nil.